scripts/tests/twister/test_quarantine.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Tests for quarantine.py classes' methods
+"""
+
+import mock
+import os
+import pytest
+
+from twisterlib.quarantine import QuarantineException, QuarantineElement, QuarantineData
+
+
+TESTDATA_1 = [
+    (
+        ['dummy scenario', 'another scenario'],
+        ['dummy platform', 'another platform'],
+        ['dummy architecture', 'another architecture'],
+        ['dummy simulation', 'another simulation'],
+        None,
+        []
+    ),
+    (
+        ['all'],
+        ['dummy platform', 'another platform'],
+        ['dummy architecture', 'another architecture'],
+        ['dummy simulation', 'another simulation'],
+        None,
+        ['scenarios']
+    ),
+    (
+        ['dummy scenario', 'another scenario'],
+        ['dummy platform', 'all'],
+        ['all', 'another architecture'],
+        ['dummy simulation', 'another simulation'],
+        None,
+        ['platforms', 'architectures']
+    ),
+    (
+        ['all', 'another scenario'],
+        [],
+        [],
+        ['all', 'all'],
+        QuarantineException,
+        ['scenarios', 'platforms', 'architectures', 'simulations']
+    ),
+]
+
+
+@pytest.mark.parametrize("scenarios, platforms, architectures, simulations, expected_exception, empty_filter_attrs", TESTDATA_1)
+def test_quarantineelement_post_init(scenarios, platforms, architectures, simulations, expected_exception, empty_filter_attrs):
```

Comment:
avoid long lines, we do not check it, but IMHO we shouldn't use more than 120ch/line